clients/client/src/main/java/org/projectnessie/client/http/NessieHttpClient.java

```diff
@@ -93,43 +91,41 @@ private static void addTracing(HttpClient.Builder httpClient) {
+    Tracer tracer = otel.getTracer("Nessie");
     if (tracer != null) {
       httpClient.addRequestFilter(
           context -> {
-            Span span = tracer.activeSpan();
-            if (span != null) {
-              Span inner = tracer.buildSpan("Nessie-HTTP").start();
-              Scope scope = tracer.activateSpan(inner);
-              context.addResponseCallback(
-                  (responseContext, exception) -> {
-                    if (responseContext != null) {
-                      try {
-                        inner.setTag(
-                            "http.status_code", responseContext.getResponseCode().getCode());
-                      } catch (IOException e) {
-                        // There's not much we can (and probably should) do here.
-                      }
+            Span span =
+                tracer
+                    .spanBuilder("Nessie-HTTP")
+                    .startSpan()
+                    .setAttribute(SemanticAttributes.HTTP_URL, context.getUri().toString())
+                    .setAttribute(SemanticAttributes.HTTP_METHOD, context.getMethod().name())
+                    .setAttribute("component", "http");
+
+            @SuppressWarnings({"resource", "MustBeClosedChecker"})
+            Scope scope = span.makeCurrent();
+
+            W3CTraceContextPropagator.getInstance()
+                .inject(Context.current(), context, RequestContext::putHeader);
+
+            context.addResponseCallback(
+                (responseContext, exception) -> {
+                  if (responseContext != null) {
+                    try {
+                      span.setAttribute(
+                          "http.status_code", responseContext.getResponseCode().getCode());
+                    } catch (IOException e) {
+                      // There's not much we can (and probably should) do here.
                     }
-                    if (exception != null) {
-                      Map<String, String> log = new HashMap<>();
-                      log.put(Fields.EVENT, Tags.ERROR.getKey());
-                      log.put(Fields.ERROR_OBJECT, exception.toString());
-                      Tags.ERROR.set(inner.log(log), true);
-                    }
-                    scope.close();
-                    inner.finish();
-                  });
-
-              inner
-                  .setTag("http.uri", context.getUri().toString())
-                  .setTag("http.method", context.getMethod().name());
-
-              HashMap<String, String> headerMap = new HashMap<>();
-              TextMap httpHeadersCarrier = new TextMapAdapter(headerMap);
-              tracer.inject(inner.context(), Builtin.HTTP_HEADERS, httpHeadersCarrier);
-              headerMap.forEach(context::putHeader);
-            }
+                  }
+                  if (exception != null) {
+                    span.setStatus(StatusCode.ERROR, exception.toString());
+                  }
+                  scope.close();
```

Comment:
Are we sure the callback is invoked on the same thread? If not, making this `scope` current (above) would be incorrect.